### DISCUSSION TITLE
How to implement incremental scope in Blazor Wasm?

### DISCUSSION THREAD

**Discussion by xperiandri** [AzureAD/microsoft-identity-web#1795] [Q&A]
I created Blazor WASM hosted on ASP.NET Core with Microsoft Identity.
It successfully works with a multi-tenant setup.
But how to handle incremental scope increase? `Team.ReadBasic.All` is what I want a consent to be provided to.
``` C#
private async Task AuthorizeTeamReadAllAsync()
{
    // https://docs.microsoft.com/en-us/aspnet/core/blazor/security/webassembly/graph-api?view=aspnetcore-6.0#graph-sdk
    var result = await TokenProvider.RequestAccessToken(
            new AccessTokenRequestOptions()
            {
                Scopes = new[] { "User.Read", "Team.ReadBasic.All", "Directory.Read.All" }
            });
}
```
@jmprieur

**Comment by jmprieur** [comment]
@xperiandri : what is TokenProvider? It's not part of Microsoft.Identity.Web?

**Reply by xperiandri** [reply]
It is from `Microsoft.Authentication.WebAssembly.Msal`

**Reply by xperiandri** [reply]
Do you have a library for WASM?

**Reply by xperiandri** [reply]
Should I move the discussion to https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/ ?
Or my request is relevant here?

**Reply by xperiandri** [reply]
If you create Blazor WASM from the template you will see
Authentication.razor
``` Razor
@page "/authentication/{action}"
@using Microsoft.AspNetCore.Components.WebAssembly.Authentication
<RemoteAuthenticatorView Action="@Action" />

@code{
    [Parameter] public string? Action { get; set; }
}
```
It handles auth with default scopes

**Reply by xperiandri** [reply]
Basically, I want that same auth popup window to open with another set of scopes

**Reply by xperiandri** [reply]
@jmprieur any updates about that?

**Comment by jmprieur** [comment]
@xperiandri blazorwasm does not rely on MSAL.NET, but MSAL.js

@javiercn would you know how to do?

**Reply by javiercn** [reply]
This is not something we currently support, but it is something we are workign on (hopefully) for .NET 7.0
